torch/jit/quantized.py

```diff
@@ -325,17 +323,19 @@ def process_weights(ihhh, layer, suffix, dtype):
                 for (ih, ih_name), (hh, hh_name) in zip(zip(ih_params, ih_param_names), zip(hh_params, hh_param_names)):
                     self.register_buffer(ih_name, torch.tensor(ih) if not isinstance(ih, torch.Tensor) else ih)
                     self.register_buffer(hh_name, torch.tensor(hh) if not isinstance(hh, torch.Tensor) else hh)
-                    self._all_weights.extend([ih_name, hh_name])
-
-        self._packed_weights = packed_weights
-        self._quantized_weights = quantized_weights
-        # For int8 quantization, _orig_weights is not needed in the quantization logic, 
-        # however there is a JIT compilation error without it. This is just used to 
-        # workaround that error. 
-        if dtype == torch.int8: 
-            self._orig_weights = self._packed_weights
-        else: 
-            self._orig_weights = orig_weights
+                    self._all_weights_names.extend([ih_name, hh_name])
+
+        # For int8 quantization, _orig_weights is not needed in the quantization logic,
+        # however there is a JIT compilation error without it. This is just used to
+        # workaround that error.
+        if dtype == torch.int8:
+            self._orig_weights_names = self._packed_weights_names
+
+        self._packed_weights = torch.jit.Attribute([getattr(self, weight) for weight in self._packed_weights_names], List[Tensor])
```

Comment:
Do we need to have torch.jit.Attribute or does it work to put the types in `__attributes__`?

Reply:
It needs to be an attribute with the current ScriptModule API, but not after my rewrite.